gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/DagProc.java

```diff
@@ -75,8 +91,19 @@ public final void process(DagManagementStateStore dagManagementStateStore,
       dagProcEngineMetrics.markDagActionsInitialize(getDagActionType(), false);
       throw e;
     }
-    act(dagManagementStateStore, state, dagProcEngineMetrics);
-    log.info("{} concluded processing for dagId : {}", getClass().getSimpleName(), this.dagId);
+    try {
+      act(dagManagementStateStore, state, dagProcEngineMetrics);
+    } catch (Exception e) {
+      if (isTransientException(e)) {
+        log.info("Ignoring transient exception. DagTask {} will conclude and will not be retried. Exception - {} ",
+            getDagTask(), e);
+        dagManagementStateStore.getDagManagerMetrics().dagProcessingNonRetryableExceptionMeter.mark();
+        dagManagementStateStore.getDagManagerMetrics().dagProcessingExceptionMeter.mark();
```

Comment:
shouldn't this line come before `if (isTransientException(e))` ?

Reply:
um, if we do that it will be double counted, because DagProcEngineThread also increases it.